tests/test_generators.py

```diff
@@ -8,58 +8,60 @@
 from deeposlandia import generator, utils
 
 
-def test_feature_detection_labelling_evaluated_labels():
-    """Test `feature_detection_labelling` function by considering only evaluated labels, *i.e.* dataset
-    labels that have a `is_evaluated` key at `False` are not integrated into the label generator
-
-    """
-    BATCH_SIZE = 5
-    MIN = 0
-    MAX = 10
-    IMAGE_SIZE = 3
-    a = np.random.randint(MIN, MAX, [BATCH_SIZE, IMAGE_SIZE, IMAGE_SIZE])
-    labels = range(MIN, MAX)
-    evaluated_labels = np.random.choice(range(MIN, MAX), MAX-3)
-    b = generator.feature_detection_labelling(a, evaluated_labels)
-    assert len(labels) != len(evaluated_labels)
-    assert b.shape == (BATCH_SIZE, len(evaluated_labels))
-
-
-def test_feature_detection_labelling():
-    """Test `semantic_segmentation_labelling` function in `generator` module:
-    - test if output shape is input shape + an additional dimension given by the `label_ids` length
-    - test if both representation provides the same information (native array on the first hand and
+def test_feature_detection_labelling_concise():
+    """Test `feature_detection_labelling` function in `generator` module by considering a concise
+    labelling, *i.e.* all labels are represented into the array:
+    * as a preliminary verification, check if passing string labels raises an AttributeError
+    exception
+    * test if output shape is first input shape (batch size) + an additional dimension given by the
+    `label_ids` length
+    * test if both representation provides the same information (native array on the first hand and
     its one-hot version on the second hand)
     """
-    BATCH_SIZE = 5
-    MIN = 0
-    MAX = 10
-    IMAGE_SIZE = 3
-    a = np.random.randint(MIN, MAX, [BATCH_SIZE, IMAGE_SIZE, IMAGE_SIZE])
-    b = generator.feature_detection_labelling(a, range(MIN, MAX))
-    assert b.shape == (BATCH_SIZE, MAX)
-
-
-def test_feature_detection_labelling_wrong_label_id():
-    """Test the value and the type of some label ids for feature detection one-hot encoding.
+    a = np.array([[[0, 0, 0, 2],
+                   [3, 3, 0, 2],
+                   [3, 3, 3, 0]],
+                  [[2, 2, 0, 0],
+                   [1, 2, 0, 0],
+                   [2, 1, 0, 0]]])
+    labels = np.unique(a).tolist()
+    MIN, MAX = np.amin(a), np.amax(a)
+    with pytest.raises(AttributeError):
```

Comment:
I didn't see that for my first review when you updated some test, but this shouldn't be an `AttributeError` exception which should be raised. It's better if we raise a `ValueError` exception. We have to update the `feature_detection_labelling` function.

Reply:
The fact is that this portion of code raises an `AttributeError` ! If we replace it with a `ValueError`, the test does not pass any more.

Reply:
That's why i wrote "we have to update the `feature_detection_labelling` function" in that way.